crates/wascc-provider/src/states/starting.rs

```diff
@@ -176,37 +167,15 @@ impl State<PodState> for Starting {
             let container_handle =
                 match start_container(pod_state, &container, &pod, port_assigned).await {
                     Ok(handle) => handle,
-                    Err(e) => {
-                        // TODO: identify if any of the start_container errors are fatal
-                        // enough that we should return an Err and exit the run loop
-                        error!("{:?}", e);
-                        let error_state = Error {
-                            message: e.to_string(),
-                        };
-                        return Ok(Transition::next(self, error_state));
-                    }
+                    Err(e) => fail_fatal!(e),
```

Comment:
I think a fatal failure here is probably not correct. In waSCC, there could be a problem with the host or lattice or a capability is not loaded. That problem could be transient or resolved by the time we come from a backoff, so this should probably just transition to an error state

Reply:
I'm wondering if this could be an opportunity for an alternative state / error message. Like  capability backoff or something.